### PR TITLE
A bit of viewgen refactor

### PR DESCRIPTION
**Pull request by elprans**
In preparation for adding support for splats, refactor some of the shape
processing bits:

1. Use named tuples for shape element descriptors
2. Factor out early processing of `ShapeElement` and pass around an
   enhanced version of it.

### CODE REVIEW

**Review by aljazerzen — APPROVED** [review]
This is going to have a few conflicts with mt rewrites, but it will also solve a few of my problems.

File: edb/edgeql/compiler/viewgen.py [hunk 1]
```diff
@@ -57,12 +58,30 @@
...
 
 if TYPE_CHECKING:
-    from edb.schema import properties as s_props
     from edb.schema import sources as s_sources
 
-    ShapePtr = Tuple[
-        irast.Set, s_pointers.Pointer, qlast.ShapeOp, Optional[irast.Set]
-    ]
+
+class ShapeElementDesc(NamedTuple):
```
Comment (msullivan) [inline edb/edgeql/compiler/viewgen.py:64]: Add some quick doc comments?

File: edb/edgeql/compiler/viewgen.py [hunk 2]
```diff
@@ -341,6 +380,94 @@ def _process_view(
     return view_scls, ir_set
 
 
+def _shape_el_ql_to_shape_el_desc(
```
Comment (msullivan) [inline edb/edgeql/compiler/viewgen.py:383]: Document what processing happens as part of this?

**Review by msullivan — APPROVED** [review]

**Comment by msullivan** [comment]
This looks great; added a couple requests for a bit more documentation, since refactoring is a great time to add more docs, since it is all fresh in mind

**Pull request merged**